lib/two_factor_authentication/models/two_factor_authenticatable.rb

```diff
@@ -73,8 +74,14 @@ def totp_enabled?
           respond_to?(:otp_secret_key) && !otp_secret_key.nil?
         end
 
+        def confirm_totp_secret(secret, code, options = {})
+          return false unless authenticate_totp(code, {otp_secret_key: secret})
+          self.otp_secret_key = secret
+          true
+        end
+
         def generate_totp_secret
-          self.otp_secret_key = ROTP::Base32.random_base32
+          ROTP::Base32.random_base32
         end
```

Comment:
Sorry, I meant the `confirm_totp_secret` method. That's the one that's not documented but needs to be called in order for this to work, since you've changed the behavior of the `generate_otp_secret` method, which no longer matches the instructions in the README. 

There are two benefits of using the database to track whether or not the otp_secret was confirmed:

1. It maintains better backwards compatibility with this gem, unlike this commit. With the first commit in this PR, the change was mostly backwards compatible. The only non-compatible change was the removal of the `before_create` callback, which meant that implementors had to either manually add a `before_create` callback to their User model that calls `generate_totp_secret`, or call `generate_totp_secret` in a controller, such as when a user is selecting their preferred 2FA options. Now, they also have to override the `generate_totp_secret` method to maintain backwards compatibility.

  I propose the following approach which minimizes changes to both existing and new apps:

  - Keep the prior version of the `generate_totp_secret` method, which sets the `otp_secret_key` value.
  - Add a `totp_confirmed_at` column to the Users table
  - Update the `authenticate_otp` method so that it sets `totp_confirmed_at` to `Time.current` if `authenticate_totp` returns true

  This means the only change necessary to existing apps is to add a migration for this new column.

2. It provides a way to query users who have tried enabling an authenticator app but never completed the process. For example, if we notice a large number of users start the process but never finish, that could be an indication of some problem and we can reach out to those users to see what happened. With the session solution, we would never know whether or not they started the process.